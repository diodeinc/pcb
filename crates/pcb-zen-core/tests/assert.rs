#[macro_use]
mod common;

// Error case â€“ evaluating `error()` should propagate the message.
snapshot_eval!(error_function_should_error, {
    "test.zen" => r#"
        error("boom")
    "#
});

// `check()` with a true condition should pass and produce a schematic/netlist.
snapshot_eval!(check_true_should_pass, {
    "test.zen" => r#"
        # check should not raise when condition is true
        check(True, "all good")
    "#
});

// `check()` with a false condition should raise and surface the message.
snapshot_eval!(check_false_should_error, {
    "test.zen" => r#"
        # check should raise when condition is false
        check(False, "failing condition")
    "#
});
