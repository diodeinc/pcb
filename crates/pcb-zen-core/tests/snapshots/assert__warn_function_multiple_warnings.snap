---
source: crates/pcb-zen-core/tests/assert.rs
expression: output
---
Warning: test.zen:2:1-22 first warning
Warning: test.zen:3:1-23 second warning
{
    <root>: Module {
        path: <root>,
        source: "test.zen",
    },
}
[]
